b2g/chrome/content/remote_control.js

```diff
@@ -20,57 +20,99 @@ const kPrefPresentationDiscoverable = "dom.presentation.discoverable";
     }
   }
 
-  function setupRemoteControl() {
-    if (Services.prefs.getBoolPref(kPrefPresentationDiscoverable)) {
-      importRemoteControlService();
-      remoteControlScope.RemoteControlService.start();
-    } else {
-      remoteControlScope.RemoteControlService.stop();
-    }
+  function getTime() {
+    return new Date().getTime();
+  }
+
+  function debug(s) {
+    let time = new Date(getTime()).toString();
+    // dump("[" + time + "] remote_control: " + s + "\n");
   }
 
-  Services.prefs.addObserver(kPrefPresentationDiscoverable, setupRemoteControl, false);
+  // Monitor the network status and the time change
+  var statusMonitor = {
+    init: function sm_init() {
+      debug("init");
+
+      Services.prefs.addObserver(kPrefPresentationDiscoverable, this, false);
+      Services.obs.addObserver(this, "network-active-changed", false);
+      Services.obs.addObserver(this, "network:offline-status-changed", false);
+      window.onmoztimechange = this.onTimeChange;
```

Comment:
Please use addEventListener, otherwise you might accidentally remove other people's timechnage listener.